lib/resty/upstream/healthcheck.lua

```diff
@@ -602,6 +606,16 @@ function _M.spawn_checker(opts)
         concurrency = concur,
     }
 
+    -- as far as we do not support https override ports if set
+    if http_port then
+        for i, _ in ipairs(ctx.primary_peers) do
+            ctx.primary_peers[i].port = http_port
```

Comment:
That is exactly what its done for. It overrides ports stated in the upstream server group, so health check can be done via another port (which is i.e. HTTP:80) in the case when upstream goes for HTTPS (in that case this plugin would no be able to do a healthcheck on it, until it is not implemented).